kubespawner/objects.py

```diff
@@ -228,7 +228,7 @@ def make_pod(
     )
 
     pod.spec = V1PodSpec(containers=[])
-    pod.spec.restartPolicy = 'Never'
+    pod.spec.restart_policy = 'OnFailure'
```

Comment:
I considered this change between restartPolicy and restart_policy, V1PodSpec - the python model representation, contains restart_policy and will render to restartPolicy later, so it makes sense. I think. Nice catch!